runtime/syntax/scad.yaml

```diff
@@ -0,0 +1,69 @@
+filetype: OpenSCAD
+
+# OpenSCAD is a functional programming language used for representing
+# 2D/3D models for use in the program of the same name.
+#
+# The following documents were used as reference material:
+# https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/The_OpenSCAD_Language
+# https://openscad.org/cheatsheet/index.html
+
+detect:
+    filename: "\\.scad$"
+
+rules:
+    - identifier: "\\b(function|module) +[a-z0-9_]+"
+
+    - statement: "\\b(abs|acos|asin|assert|atan|atan2|ceil)\\b"
+    - statement: "\\b(child|children|chr|circle|color)\\b"
+    - statement: "\\b(concat|cos|cross|cube|cylinder)\\b"
+    - statement: "\\b(difference|dxf_cross|dxf_dim|each)\\b"
+    - statement: "\\b(echo|else|exp|floor|for|function|hull)\\b"
+    - statement: "\\b(if|import|import_dxf|intersection)\\b"
+    - statement: "\\b(intersection_for|is_bool|is_function)\\b"
+    - statement: "\\b(is_list|is_num|is_string|is_undef|len)\\b"
+    - statement: "\\b(let|linear_extrude|ln|log|lookup|max)\\b"
+    - statement: "\\b(min|minkowski|mirror|module|multmatrix)\\b"
+    - statement: "\\b(norm|offset|ord|parent_module|polygon)\\b"
+    - statement: "\\b(polyhedron|pow|projection|rands|render)\\b"
+    - statement: "\\b(resize|rotate|rotate_extrude|round)\\b"
+    - statement: "\\b(scale|search|sign|sin|sphere|sqrt)\\b"
+    - statement: "\\b(square|str|surface|tan|text|translate)\\b"
+    - statement: "\\b(union|version|version_num)\\b"
```

Comment:
I know it doesn't help for readability, but it should slightly improve the performances when we merge patterns of the same type.
The highlighter mechanism then doesn't need to cycle more rounds and can identify the type by one single and unfortunately large regex rule.

Maybe this is valid for more rules here.

In case some rules should be differentiated, but are of the same type, then they can be extended with a subgroup...as long as this subgroup type (e.g. `statement.foo` & `statement.bar`) doesn't have another different color defined within the color scheme (see here: [colors.md](https://github.com/zyedidia/micro/blob/master/runtime/help/colors.md)).

Reply:
OK, that makes sense.  Sounds like I should at least merge the statement rules, and I think I should also merge the constant.number rules.  I will do that.

Any suggestions re best practices for updating the PR?  Do I just push a new commit to my PR branch?  Or should I flatten the two commits into one?  Or make a new branch with the two commits flattened into one and then update or replace the pull request using that branch?  Sorry to ask such a basic question...

Reply:
> Any suggestions re best practices for updating the PR? Do I just push a new commit to my PR branch? Or should I flatten the two commits into one?

Since it's a fully new file anyway there is no problem to provide it in one single/squashed commit to the same branch, from which the PR was created. The rest will be done by GitHub.

Reply:
OK, I updated per the feedback, rebuilt, tested, and pushed the revised commit.
The github GUI seems to show the revised commit now.

LMK if I missed anything, and thanks for the help.